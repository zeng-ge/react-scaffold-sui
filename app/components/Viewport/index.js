import React, { Component } from 'react'
import { array, object, oneOfType, string } from 'prop-types'
import classnames from 'classnames'
import less from './index.less'

export default class extends Component {

  static propTypes = {
    children: oneOfType([array, object]).isRequired,
    className: string,
  }

  render() {
    const { children, className = '', ...restProps } = this.props
    return (
      <div className={classnames(less.viewport, className)} {...restProps}>
        { children }
      </div>
    )
  }
}
