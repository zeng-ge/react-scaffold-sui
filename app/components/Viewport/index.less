@import '../../styles/mixins.less';
.viewport{
  display: flex;
  flex-flow: column nowrap;
  .viewport();
}
