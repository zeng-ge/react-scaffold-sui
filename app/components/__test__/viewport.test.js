import React from 'react'
import render from 'react-test-renderer'
import Viewport from '../Viewport'
import { shallow } from 'enzyme'

describe('Viewport', () => {
  test('viewport snapshot', () => {
    const viewport = render.create(
      <Viewport className="viewport" id="firstViewport">
        <div></div>
      </Viewport>
    )
    expect(viewport.toJSON()).toMatchSnapshot()
  })

  test('shoud viewport contains assigned child', () => {
    const component = shallow(
      <Viewport>
        <div>abc</div>
      </Viewport>
    )
    expect(component.contains(<div>abc</div>)).toBeTruthy()
  })

})
