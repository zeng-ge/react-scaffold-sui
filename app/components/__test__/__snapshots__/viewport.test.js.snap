// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewport viewport snapshot 1`] = `
<div
  className="viewport viewport"
  id="firstViewport"
>
  <div />
</div>
`;
